FGEGraphics/GraphicsHelpers/Models/ShapeGenerators.cs

```diff
@@ -0,0 +1,408 @@
+        uint vertexCount = (rings + 1) * (sides + 1);
+        uint numIndices = rings * sides * 6;  // 2 triangles per ring/side, 3 indices per triangle
+
+        List<Vector3> vertices = new((int)vertexCount);
+        List<Vector3> normals = new((int)vertexCount);
+        List<Vector2> textureCoords = new((int)vertexCount);
+
+        uint[] indices = new uint[numIndices];
+        int indexIndex = 0;
+
+        for (int i = 0; i <= rings; i++)
+        {
+            float theta = i * MathHelper.TwoPi / rings;
+            float sinTheta = (float)Math.Sin(theta);
+            float cosTheta = (float)Math.Cos(theta);
+
+            for (int j = 0; j <= sides; j++)
+            {
+                float phi = j * MathHelper.TwoPi / sides;
+                float sinPhi = (float)Math.Sin(phi);
+                float cosPhi = (float)Math.Cos(phi);
+
+                float centerX = radius * cosPhi;
+                float centerY = 0;
+                float centerZ = radius * sinPhi;
+
+                float x = (radius + tubeRadius * cosTheta) * cosPhi;
+                float y = tubeRadius * sinTheta;
+                float z = (radius + tubeRadius * cosTheta) * sinPhi;
+
+                vertices.Add(new Vector3(x, y, z));
+
+                Vector3 pointOnSurface = new(x, y, z);
+                Vector3 centerToSurface = pointOnSurface - new Vector3(centerX, centerY, centerZ);
+                normals.Add(centerToSurface.Normalized());
+
+                textureCoords.Add(new Vector2((float)j / sides, (float)i / rings));
+
+                if (i < rings && j < sides)
+                {
+                    uint currentRow = (uint)(i * (sides + 1));
+                    uint nextRow = (uint)((i + 1) * (sides + 1));
+
+                    indices[indexIndex++] = (uint)(currentRow + j);
+                    indices[indexIndex++] = (uint)(nextRow + j);
+                    indices[indexIndex++] = (uint)(nextRow + j + 1);
+
+                    indices[indexIndex++] = (uint)(currentRow + j);
+                    indices[indexIndex++] = (uint)(nextRow + j + 1);
+                    indices[indexIndex++] = (uint)(currentRow + j + 1);
+                }
+            }
+        }
+
+        if (vertices.Count != normals.Count || vertices.Count != textureCoords.Count)
+            throw new ArgumentException("Position, normal, and texture coordinate lists must have the same length.");
+
+        if (reverseOrder)
+        {
+            vertices.Reverse();
+            textureCoords.Reverse();
+            normals.Reverse();
+            Array.Reverse(indices);
+        }
+
+        return GetModelAfterGenerating(modelEngine, "torus", vertices, normals, textureCoords, indices);
+    }
+
+    private static Model GetModelAfterGenerating(
+        ModelEngine engine,
+        string name,
+        List<Vector3> vertices,
+        List<Vector3> normals,
+        List<Vector2> texCoords,
+        uint[] indices)
```

Comment:
wtf

Reply:
It was getting long so thought this would be a good way to make sure I can see all the parameters if I recall I saw this done somewhere in the System namespace but will change.